blockchain/tests/block_production.rs

```diff
@@ -1377,7 +1377,7 @@ fn it_can_revert_basic_and_create_contracts_txns() {
 }
 
 fn ed25519_key_pair(secret_key: &str) -> SchnorrKeyPair {
-    let priv_key: SchnorrPrivateKey =
-        Deserialize::deserialize_from_vec(&hex::decode(secret_key).unwrap()[..]).unwrap();
+    let priv_key =
+        SchnorrPrivateKey::deserialize_from_vec(&hex::decode(secret_key).unwrap()).unwrap();
```

Comment:
I see that you now changed all deserializations to use the struct, which I like! But is there anything safeguarding against writing it the old way in the future? Can we add a custom clippy rule for this? Otherwise this won't necessarily stick.

Reply:
I haven't replaced all of the `Deserialize::deserialize` calls, only those where it wasn't difficult to specify.

I don't know about custom clippy lints, but it seems like a good idea in general. ^^

Reply:
For future reference: https://github.com/trailofbits/dylint